obspy/io/hypodd/pha.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+"""
+HypoDD PHA read support.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+import io
+
+from obspy import UTCDateTime
+from obspy.core.event import (
+    Catalog, Event, Origin, Magnitude, Pick, WaveformStreamID, Arrival,
+    OriginQuality)
+
+
+def _seed_id_map(inventory=None, id_map=None, id_default='.{}..{}'):
+    if id_map is None:
+        id_map = {}
+    ret = id_map.copy()
+    if inventory is not None:
+        for net in inventory:
+            for sta in net:
+                if len(sta) == 0:
+                    temp = id_map.get(sta.code, id_default)
+                    temp = temp.split('.', 2)[-1]
+                else:
+                    cha = sta[0]
+                    temp = cha.location_code + '.' + cha.code[:-1] + '{}'
+                ret[sta.code] = net.code + '.{}.' + temp
+    return ret
+
+
+def _block2event(block, seed_map, id_default, ph2comp):
+    """
+    Read HypoDD event block
+    """
+    lines = block.strip().splitlines()
+    yr, mo, dy, hr, mn, sc, la, lo, dp, mg, eh, ez, rms, id_ = lines[0].split()
+    time = UTCDateTime('{}-{}-{} {}-{}-{}'.format(yr, mo, dy, hr, mn, sc))
```

Comment:
```suggestion
    time = UTCDateTime(int(yr), int(mo), int(dy), int(hr), int(mn), float(sc))
```

Okay - seeing it written out I'm no longer sure this is cleaner.

Reply:
I think its a bit cleaner...